analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -0,0 +1,156 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Protobuf;
+using SonarAnalyzer.Rules;
+using SonarAnalyzer.UnitTest.TestFramework;
+using CS = SonarAnalyzer.Rules.CSharp;
+using VB = SonarAnalyzer.Rules.VisualBasic;
+
+namespace SonarAnalyzer.UnitTest.Rules
+{
+    [TestClass]
+    public class SymbolReferenceAnalyzerTest
+    {
+        private const string Root = @"TestCases\Utilities\SymbolReferenceAnalyzer\";
+
+        public TestContext TestContext { get; set; } // Set automatically by MsTest
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Method_PreciseLocation_CS() =>
+            Verify("Method.cs", references =>
+            {
+                references.Select(x => x.Declaration.StartLine).Should().BeEquivalentTo(1, 3, 5);   // class 'Sample' on line 1, method 'Method' on line 3, method 'Go' on line 5
+                var methodDeclaration = references.Single(x => x.Declaration.StartLine == 3);
+                methodDeclaration.Declaration.Should().BeEquivalentTo(new TextRange { StartLine = 3, EndLine = 3, StartOffset = 16, EndOffset = 22 });
+                methodDeclaration.Reference.Should().HaveCount(1);
+                methodDeclaration.Reference.Single().Should().BeEquivalentTo(new TextRange { StartLine = 6, EndLine = 6, StartOffset = 8, EndOffset = 14 });
+            });
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Method_PreciseLocation_VB() =>
+            Verify("Method.vb", references =>
+            {
+                references.Select(x => x.Declaration.StartLine).Should().BeEquivalentTo(1, 3, 6);   // class 'Sample' on line 1, method 'Method' on line 3, method 'Go' on line 6
+                var methodDeclaration = references.Single(x => x.Declaration.StartLine == 3);
+                methodDeclaration.Declaration.Should().BeEquivalentTo(new TextRange { StartLine = 3, EndLine = 3, StartOffset = 15, EndOffset = 21 });
+                methodDeclaration.Reference.Should().HaveCount(1);
+                methodDeclaration.Reference.Single().Should().BeEquivalentTo(new TextRange { StartLine = 7, EndLine = 7, StartOffset = 8, EndOffset = 14 });
+            });
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Event_CS() =>
+            Verify("Event.cs", 6, 5, 9, 10);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Field_CS() =>
+            Verify("Field.cs", 4, 3, 7, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Local_CS() =>
+            Verify("Local.cs", 4, 7, 5);
```

Comment:
It would be better to rename Local to LocalFunction or something similar to avoid confusion.